### ISSUE TITLE
JFR Image Size: Support reading plugin manifest files for plugins in ClassPath

### ISSUE THREAD

**Issue by oleg-nenashev** [jenkinsci/jenkinsfile-runner#390]
[JFR Packaging POM](https://github.com/jenkinsci/jenkinsfile-runner/tree/master/packaging-parent-pom) currently prepares a Jenkinsfile Runner bundle (appassembler ZIP) where plugins are included into the bundle. They are available in the classpath, but JFR Plugin Manager cannot recognize the plugins because of the missing plugin Manifest resources. After that, API like `PluginManager#getPlugin(String)` do not work properly and cause random issues in plugins. It would be nice to have a way to read proper manifests of plugins included into classpath.

This is a second stage of #389 which makes JFR to consume the file. This is also an obstacle I referenced in #368 from @PierreBtz 

## Acceptance Criteria

* Jenkinsfile Runner's Plugin Manager implementation is updated to support reading the manifest file generated in #389. The Plugin Manager reads the Manifest file on the JFR startup and initializes the loaded plugin list accordingly (passed by option or a specific resource location)
  * Taking [JENKINS-48885](https://issues.jenkins-ci.org/browse/JENKINS-48885), it might make sense to implement this code in a library or the Jenkins Core  so that both Jenkinsfile Runner and Jenkins Test Harness can use the same codebase.
* Initialization happens before `--plugins` argument is processed. It will also help to discover conflicts between plugins in the classpath and what is supplied by the plugins file
* `Vanilla` and `App` packaging in this repository are updated to use the plugin manifest file
* There is test coverage which verifies that the desired plugins were loaded by Jenkinsfile Runner

## Notes

Depends on #389

**Comment by oleg-nenashev** [comment]
https://github.com/oleg-nenashev/lib-bom-jep309/pull/1 as a foundation lib
